File: Solutions/Bulls_and_Cows/Bulls_and_Cows.js
//One pass solution
var getHint = function(secret, guess) {
    let bull = 0;
    let cow = 0;
    const numCount = new Array(10).fill(0);
    
    for(let i = 0; i < secret.length; i++) {
        if(secret[i] === guess[i]) bull++;
        else {
            numCount[+secret[i]]++;
            if(numCount[+secret[i]] <= 0) cow++;
            numCount[+guess[i]]--;
            if(numCount[+guess[i]] >= 0) cow++;
        }
    }
    return `${bull}A${cow}B`
};

//Fastest
const getHint = (secret, guess) => {
    let bulls = 0, cows = 0, numbers = new Array(10).fill(0);
    
    for (let i = 0; i<secret.length; i++) {
        if (secret.charAt(i) == guess.charAt(i)) bulls++;
        else {
            if (numbers[+secret.charAt(i)]++ < 0) cows++;
            if (numbers[+guess.charAt(i)]-- > 0) cows++;
        }
    }
    return bulls + "A" + cows + "B";
};

File: Solutions/Bulls_and_Cows/Bulls_and_Cows.py
class Solution:
    def getHint(self, secret, guess):
        B = sum([x==y for x,y in zip(secret, guess)])
        Count_sec = Counter(secret)
        Count_gue = Counter(guess)
        B_C = sum([min(Count_sec[elem], Count_gue[elem]) for elem in Count_sec])
        return str(B) + "A" + str(B_C-B) + "B"

#Fastest
class Solution:
    def getHint(self, secret: str, guess: str) -> str:
        
        bull=0
        for i in range(len(secret)):
            bull += int(secret[i] == guess[i])
        
		# This loop will take care of "cow" cases
        cows=0
        for c in set(secret):
            cows += min(secret.count(c), guess.count(c))
        
        return f"{bull}A{cows-bull}B"